### ISSUE TITLE
Consider opinion by Jeff

### ISSUE THREAD

**Issue by quinnliu** [WalnutiQ/WalnutiQ#181]
<b>Next Action:</b> Consider the following:

The question of whether deep learning or HTM  is further along towards machine intelligence is an interesting one. I am often asked about deep learning and variations of this question.  Here is how I answer it.
 
Machine intelligence requires a number of attributes in a learning system.  Here are my top four attributes, there are more, but this is a good start.
1) Continuous learning.
2) Learning and inferring from time-based data streams.
3) Integration of sensory data and motor behavior.
4) Hierarchy
 
Hierarchy is needed to scale the system to handle larger sensory arrays, multiple sensory arrays, and deeper temporal contexts. But I don’t believe it adds any fundamental new properties.  That is, once we know how to build a two level hierarchy we will know how to build a ten level hierarchy.  At Numenta we have been working on understanding how a single cortical region works.  We understand how real neurons learn, infer, and predict from time-based data streams. We understand how they learn continuously.  We are currently working on sensory motor integration issues, i.e. how we learn through movement and how we create new behaviors. We have postponed work on hierarchy until we understand the rest.  That is the approach we are taking. Time will tell if that is the best approach or not.
 
Most deep learning networks today don’t incorporate any of the first three principles but they do have the fourth. My fear is that most deep learning researchers do not understand that they need to add the other attributes and they will get stuck in a local minimum. Some people are starting to add time to dl networks and some are trying to do robotics with dl networks but from what I see they are not going about this in a principled way. Time will tell if they are or not.  Either way the vast majority of dl researchers don’t appear to understand the importance of sensory motor integration, time-based inference, and continuous learning.
 
The world will end up with a single set of design principles for intelligent machines.  Every one working on machine intelligence wants to get to that end point, but people disagree on the best way to get there. To me, deep learning and HTMs are not alternate ideas, they are  just different starting points and different philosophies on how best to get to the same endpoint. It is too early to say which approach is ahead of the other and whether dl networks will get stuck in a local minimum.
 
I just finished David McCullough’s book about the Wright Brothers. It was a fascinating. The Wright brothers decided that the key problem with human flight was how to control the movement of the plane. For years they worked on this problem, mostly with gliders. Only after they figured out control did they bother to add an engine, and even then they focused on improving lift and control. They were patient and methodical. Other pioneers took a different approach. They started with gliders and immediately put engines on them. Their planes flew but were not easily controlled. The Wright brothers are credited with the invention of the airplane not because they were the first to fly, but because they were the first to build an airplane that could be safely and accurately controlled. Other early planes took off and landed. The Wright brother’s planes flew gracefully, made accurate turns, landed precisely, etc.
 
From my perspective, HTM research is approaching the goal of machine intelligence with a more patient and methodical approach. I feel confident that the principles we are working on  will be essential in the final solution.  Whether it is the fastest approach to the end goal we won’t know until after the fact.
Jeff

**Comment by quinnliu** [comment]
If the MARK II experiment_1 doesn't work focus on creating an incrediabley simple to understand experiment with 1 region running SP and TP
